runhttpd.sh

```diff
@@ -2,11 +2,11 @@
 
 PROVISIONING_INTERFACE=${PROVISIONING_INTERFACE:-"provisioning"}
 HTTP_PORT=${HTTP_PORT:-"80"}
-HTTP_IP=$(ip -4 address show dev "$PROVISIONING_INTERFACE" | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | head -n 1)
+HTTP_IP=$(ip -br addr show dev $PROVISIONING_INTERFACE | grep -Po "[^\s]+/[0-9]+" | grep -e "^fd" -e "\." | sed -e 's%/.*%%' | head -n 1)
```

Comment:
I think we need to replace the `grep -Po "[^\s]+/[0-9]+"` because the minikube busybox shell doesn't support the `-P` option, and also I think that numeric regex is now wrong for the hex ipv6 addresses?

Reply:
> I think we need to replace the `grep -Po "[^\s]+/[0-9]+"` because the minikube busybox shell doesn't support the `-P` option, 

I can submit an alternative without the -P, is this scripts being used somewhere outside of the container is being built in?

> and also I think that numeric regex is now wrong for the hex ipv6 addresses?

I'm not sure what you mean here which regex is wrong?


> @hardys hardys referenced this pull request 28 minutes ago
> Listen IPv6 addresses for HTTPD #107

Sorry I hadn't seen this had already started in another PR, I can abandon if ye want to continue with the other approach.

Reply:
Ah sorry, I misread the regex - it's fine as it's matching string/NN - the reason I hit the -P issue is I tried testing it via copy/paste on busybox, so that's probably not an issue.

Re the other PR there has been some good discussion there so lets see if we can reach conclusion re the approach there then update whichever PR makes sense